### PR TITLE
prometheus: scrapers are not stopped when receiver is shutdown

### PR DESCRIPTION
**Pull request by jrcamp**
During receiver shutdown discovery is stopped. However, any previously
discovered targets will continue to be scraped. This calls `Stop` on the scrape
manager to stop the scraping on shutdown as well.

**Issue:** https://github.com/signalfx/splunk-otel-collector/issues/471

**Testing:** 
- added check to Prometheus end to end test that scraping is actually stopped
- tested end to end with receivercreator,  saw that scraping actually stopped after pod was deleted

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
Build failed.

File: receiver/prometheusreceiver/metrics_receiver_test.go [hunk 1]
```diff
@@ -1400,6 +1406,22 @@ func testEndToEnd(t *testing.T, targets []*testData, useStartTimeMetric bool) {
 			target.validateFunc(t, target, results[target.name])
 		})
 	}
+
+	// verify state after shutdown is called
+	assert.Lenf(t, flattenTargets(rcvr.scrapeManager.TargetsAll()), 3, "expected 3 targets to be running")
+	shutdownCalled = true
+	require.NoError(t, rcvr.Shutdown(context.Background()))
+
+	assert.Len(t, flattenTargets(rcvr.scrapeManager.TargetsAll()), 0, "expected scrape manager to have no targets")
```
Comment (dashpole) [inline receiver/prometheusreceiver/metrics_receiver_test.go:1415]: should we just put this in the cleanup function, and remove the Shutdown call here?

**Review by dashpole — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: receiver/prometheusreceiver/metrics_receiver_test.go (hunk 1 above)
Comment (jrcamp) [reply receiver/prometheusreceiver/metrics_receiver_test.go:1415]: Yeah that's much cleaner, wasn't sure if it was ok to put assertions in the Cleanup handler but seems to work fine.

**Comment by jrcamp** [comment]
@tigrannajaryan tests are fixed. Fixing the stopping surfaced a deadlock that's now been fixed.

**Comment by punya** [comment]
@Aneurysm9 @dashpole could you take another look, now that tests are fixed?

**Review by dashpole — APPROVED** [review]
Still looks good!

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**